module_fcst_grid_comp.F90

```diff
@@ -201,7 +203,37 @@ subroutine SetServicesNest(nest, rc)
       grid_typekind = ESMF_TYPEKIND_R8
     endif
 
-    if (trim(name)=="global") then
+    if (Atmos%grid_type==4) then
+       gridfile = "grid_spec.nc" ! default
+
+       if (open_file(fileobj, "INPUT/grid_spec.nc", "read")) then
+         if (variable_exists(fileobj, "atm_mosaic_file")) then
+           call read_data(fileobj, "atm_mosaic_file", gridfile)
+         endif
+         call close_file(fileobj)
+       endif
```

Comment:
Looks to me `gridfile` is not used anymore, other than in the error message below. If it's not, it should be removed.

Reply:
This block has been removed.